collector/multiplier.go

```diff
@@ -0,0 +1,72 @@
+package collector
+
+import (
+	"context"
+	"sync"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type Config struct {
+	Collectors []Interface
+	Logger     micrologger.Logger
+}
+
+type Multiplier struct {
+	collectors []Interface
+	logger     micrologger.Logger
+
+	bootOnce sync.Once
+}
+
+func New(config Config) (*Multiplier, error) {
+	if len(config.Collectors) == 0 {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Collectors must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	m := &Multiplier{
+		collectors: config.Collectors,
+		logger:     config.Logger,
+
+		bootOnce: sync.Once{},
+	}
+
+	return m, nil
+}
+
+func (m *Multiplier) Boot(ctx context.Context) {
+	m.logger.Log("level", "debug", "message", "booting collector")
+
+	m.bootOnce.Do(func() {
+		for _, c := range m.collectors {
+			c.Boot(ctx)
```

Comment:
I would actually like to have the `Boot` interface everywhere to return errors but I am not sure how to implement it so I need some feedback. I was thinking instead of using `sync.Once` we could do some channel magic and stay in the same scope of the function body. We maybe could have some tiny implementation like `github.com/giantswarm/sync`.